cmd/exchange_template/exchange_template.go

```diff
@@ -130,112 +126,111 @@ func main() {
+			FilePostfix:  ".go",
+			TemplateFile: "main_file.tmpl",
+		},
+		{
+			Name:         "test",
+			Filename:     "test_file.tmpl",
+			FilePostfix:  "_test.go",
+			TemplateFile: "test_file.tmpl",
+		},
+		{
+			Name:         "type",
+			Filename:     "type_file.tmpl",
+			FilePostfix:  "_types.go",
+			TemplateFile: "type_file.tmpl",
+		},
+		{
+			Name:         "wrapper",
+			Filename:     "wrapper_file.tmpl",
+			FilePostfix:  "_wrapper.go",
+			TemplateFile: "wrapper_file.tmpl",
+		},
 	}
 
-	exchangeDirectory = filepath.Join(exchangePackageLocation, newExchangeName)
-	exchangeTest = filepath.Join(exchangeDirectory, fmt.Sprintf(packageTests, newExchangeName))
-	exchangeTypes = filepath.Join(exchangeDirectory, fmt.Sprintf(packageTypes, newExchangeName))
-	exchangeWrapper = filepath.Join(exchangeDirectory, fmt.Sprintf(packageWrapper, newExchangeName))
-	exchangeMain = filepath.Join(exchangeDirectory, fmt.Sprintf(packageMain, newExchangeName))
-	exchangeReadme = filepath.Join(exchangeDirectory, packageReadme)
-
-	err = os.Mkdir(exchangeDirectory, 0700)
+	exchangeDirectory := filepath.Join(targetPath, exch.Name)
+	err = os.MkdirAll(exchangeDirectory, 0770)
 	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool cannot make directory ", err)
+		return err
 	}
 
-	tReadme, err := template.New("readme").ParseFiles("readme_file.tmpl")
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool error ", err)
-	}
-	newFile(exchangeReadme)
-	r1, err := os.OpenFile(exchangeReadme, os.O_WRONLY, 0700)
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool cannot open file ", err)
-	}
-	tReadme.Execute(r1, exch)
-	r1.Close()
+	fmt.Printf("Output directory: %s\n", exchangeDirectory)
+	for x := range outputFiles {
+		var tmpl *template.Template
+		tmpl, err = template.New(outputFiles[x].Name).ParseFiles(outputFiles[x].TemplateFile)
+		if err != nil {
+			return fmt.Errorf("%s template error: %s", outputFiles[x].Name, err)
+		}
 
-	tMain, err := template.New("main").ParseFiles("main_file.tmpl")
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool error ", err)
-	}
-	newFile(exchangeMain)
-	m1, err := os.OpenFile(exchangeMain, os.O_WRONLY, 0700)
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool cannot open file ", err)
-	}
-	tMain.Execute(m1, exch)
-	m1.Close()
+		filename := outputFiles[x].Filename
+		if outputFiles[x].FilePostfix != "" {
+			filename = exch.Name + outputFiles[x].FilePostfix
+		}
 
-	tTest, err := template.New("test").ParseFiles("test_file.tmpl")
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool error ", err)
-	}
-	newFile(exchangeTest)
-	t1, err := os.OpenFile(exchangeTest, os.O_WRONLY, 0700)
-	if err != nil {
-		log.Fatal("GoCryptoTrader: Exchange templating tool cannot open file ", err)
-	}
-	tTest.Execute(t1, exch)
-	t1.Close()
+		fmt.Printf("Output template name: %s template file: %s filename: %s",
```

Comment:
should we add a new line here ?
```sh
without:
❯ go run . -name "hellowrld" -rest -ws
GoCryptoTrader: Exchange templating tool.
Copyright (c) 2014-2020 The GoCryptoTrader Developers.

Exchange Name:  hellowrld
Websocket Supported:  true
REST Supported:  true
FIX Supported:  false

Please check if everything is correct and then type y to continue or n to cancel...
y
exchange ITBIT last manual update of available currency pairs has exceeded 30 days. Manual update required!
Output directory: ../../exchanges/hellowrld
Output template name: readme template file: readme_file.tmpl filename: README.mdOutput template name: main template file: main_file.tmpl filename: hellowrld.goOutput template name: test template file: test_file.tmpl filename: hellowrld_test.goOutput template name: type template file: type_file.tmpl filename: hellowrld_types.goOutput template name: wrapper template file: wrapper_file.tmpl filename: hellowrld_wrapper.goGoCryptoTrader: Exchange templating tool service complete
When the exchange code implementation has been completed (REST/Websocket/wrappers and tests), please add the exchange to engine/exchange.go
Add the exchange config settings to config_example.json (it will automatically be added to testdata/configtest.json)
Increment the available exchanges counter in config/config_test.go
Add the exchange name to exchanges/support.go
Ensure go test ./... -race passes
Open a pull request
If help is needed, please post a message in Slack.
```

Reply:
Will do!